docs/reference/ilm/policy-definitions.asciidoc

```diff
@@ -0,0 +1,293 @@
+[role="xpack"]
+[testenv="basic"]
+[[index-lifecycle-management]]
+= Policy Phases and Actions
+
+There are four stages in the index lifecycle:
+
+* Hot--the index is actively being updated and queried.
+* Warm--the index is no longer being updated, but is still being queried.
+* Cold--the index is no longer being updated and is seldom queried. The
+information still needs to be searchable, but it's okay if those queries are
+slower.
+* Delete--the index is no longer needed and can safely be deleted.
+
+Each of these stages is called a "phase". A policy does not need to configure
+each phase for an index. For example, one index may only go from the hot phase to
+the delete phase, while another may run through all phases. Depending on which
+phase one is in, different actions are allowed to be taken.
+
+== Actions
+
+Before discussing which actions are allowed in which phase, it is important
+to understand all the actions as they are.
+
+[[ilm-allocate-action]]
+=== Allocate
+
+The Allocate action allows you to specify which nodes are allowed to host the
+shards of the index and set the number of replicas.
+Behind the scenes, it is modifying the index settings
+for shard filtering and/or replica counts. When updating number of replicas,
+configuring allocation rules is optional. When configuring allocation rules,
+setting number of replicas is optional. Although this action can be treated as
+two separate index settings updates, both can be configured at once.
+
+Read more about index replicas <<getting-started-shards-and-replicas,here>>.
+Read more about shard allocation filtering in
+the <<shard-allocation-filtering,Shard allocation filtering documentation>>.
+
+[[ilm-allocate-options]]
+.Allocate Options
+[options="header"]
+|======
+| Name                 | Required  | Default     | Description
+| `number_of_replicas` | no        | -           | The number of replicas to assign to the index
+| `include`            | no        | -           | assigns an index to nodes having at least _one_ of the values
+| `exclude`            | no        | -           | assigns an index to nodes having _none_ of the values
+| `require`            | no        | -           | assigns an index to nodes having _all_ of the values
+|======
+
+Although all configuration options are not required on their own. If `number_of_replicas` is not
+configured, then at least one of `include`, `exclude`, and `require` is required. An empty
+Allocate Action with no configuration is invalid.
+
+==== Example: Change number of replicas
+
+In this example, the index's number of replicas is changed to `2`, while allocation
+rules are unchanged.
+
+[source,js]
+--------------------------------------------------
+{
+  "allocate" : {
+    "number_of_replicas" : 2
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
```

Comment:
Would we prefer real CONSOLE examples here? I hesitated at first due to the extra noise, but I also see the value of running these snippets and giving further context about which phases these actions are defined in. What do you think?

Reply:
I agree with @dakrone. Also having NOTCONSOLE examples means we may miss places to update the documentation if/when things change in the future